pkg/components/overrides.go

```diff
@@ -55,3 +55,9 @@ type ReplicatedPodResourceOverrides interface {
 
 	GetDeploymentStrategy() *appsv1.DeploymentStrategy
 }
+
+type PodResourceOverrides interface {
```

Comment:
Is there a reason we needed to introduce a new interface for this?

As far as I can tell, `ReplicatedPodResourceOverrides` should do the trick. Sure, some of the fields (GetDeploymentStrategy) don't make sense for a job. But we just return `nil` for the methods that don't apply and they are skipped.

Reply:
K8S Jobs do not have `MinReadySeconds` and `DeploymentStrategy`. Altough, the rest of the methods are applicable to a resource that contains PodTemplate objects.